docs/api-reference/core/transform/texture-transform.md

```diff
@@ -0,0 +1,67 @@
+# TextureTransform
+
+`TextureTransform` is an internal helper class for `Transform`, responsible for managing resources and state required for reading from and/or writing to `Texture` objects. It auto creates `Texture` objects when requested, creates `Framebuffer` objects. Maintains all texture bindings, when swapping is eanbled, two binding objects are created for easy switching of all WebGL resource binginds.
+
+NOTE: In following sections 'texture transform' is used to refer to 'reading from and/or writing to `Texture` objects'.
+
+## Constructor
+
+### Transform(gl : WebGL2RenderingContext, props: Object)
+
+* `gl` (`WebGLRenderingContext`) gl - context
+* `props` (`Object`, Optional) - contains following data.
+  * `sourceTextures` (`Object`, Optional) - key and value pairs, where key is the name of vertex shader attribute and value is the corresponding `Texture` object.
+  * `targetTexture` (`Texture`|`String`, Optional) - `Texture` object to which data to be written. When it is a `String`, it must be one of the source texture attributes name, a new texture object is cloned from it.
+  * `targetTextureVarying` (`String`) : varying name used in vertex shader who's data should go into target texture.
+  * `swapTexture` (`String`) : source texture attribute name, that is swapped with target texture every time `swap()` is called.
+  * `fs` (`String`, Optional) - fragment shader string, when rendering to a texture, fragments can be processed using this custom shader, when not specified, pass through fragment shader will be used.
+
+
+## Methods (Model props)
+
+### getDrawOptions(opts: Object) : Object
```

Comment:
I feel, `updateDrawOptions` is more suitable, here and for `BufferTransform`.

Reply:
It's quite awkward how the props/options are passed from the parent class to the child then back again. Shouldn't `BufferTransform` and `TextureTransform` own their models instead of `Transform`?

Reply:
as discussed offline, these are internal methods to Transform, I will address these concerns in Phase-2, where I am planning to make this API public.